x-pack/plugins/alerting/server/monitoring/rule_last_run_service.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { getDefaultLastRun } from '../lib/last_run';
+import { AlertsCount, PublicLastRunSetters, RuleLastRun, RuleLastRunOutcomes } from '../types';
+
+export class RuleLastRunService {
+  private lastRun: RuleLastRun = getDefaultLastRun();
+  private shouldOverrideFrameworkLastRun: boolean = false;
```

Comment:
I might have agree to early on this one, @mikecote brought a good point here that it is possible that the framework need to overwrite when for example we can not enqueue the rule actions. So we might still need to by pass your suggested outcome.